tests/unit/s2n_connection_test.c

```diff
@@ -124,7 +124,7 @@ int main(int argc, char **argv)
      */
     {
         /* Carefully consider any increases to this number. */
-        const uint16_t max_connection_size = 4250;
+        const uint16_t max_connection_size = 4264;
```

Comment:
`struct s2n_connection;` contains one member value of `struct s2n_kex_parameters`, which contains two seperate `struct s2n_kem_group_params;`, one for the server-side and a 2nd for the client-side:

https://github.com/aws/s2n-tls/blob/ecedd2b2c217cda4d7833c1f25a5e51f8a54c655/tls/s2n_crypto.h#L37-L38

So theoretically, my change should increase the size of an `s2n_connection` by at least 2 bits. Since there aren't any other bitfield members available in the structs that I added that bit to, each new bit is rounded up to 1 whole byte. Then on top of that, [each struct member is usually required to be native-word aligned](https://stackoverflow.com/a/119134) so that it begins on a (64-bit/8-byte) boundary, so if that 1-byte of extra memory in `server_kem_group_params` and `client_kem_group_params` crosses over into the next word, each struct gets padded out to 8 bytes, resulting in a ~16-byte increase to the size of s2n_connection.

Reply:
This would be a (very small, granted) argument for tracking len_prefixed once on the connection, where there's already a bitfield.